### PR TITLE
azure/stack: new module

### PR DESCRIPTION
**Pull request by samkahchiin**

### CODE REVIEW

File: azure/virtual-network/db-subnet.tf [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    destination_port_range     = 5432
+    source_address_prefix      = "*"
+    destination_address_prefix = "*"
+  }
+
+  tags = coalesce(var.tags, local.default_tags)
+}
+
+resource "azurerm_subnet" "db" {
+  name                 = "${local.default_name}-db-subnet"
```
Comment (swiknaba) [inline azure/virtual-network/db-subnet.tf:22]: pls fix faulty merge, this file included fixes that haven't been merged correctly: https://github.com/dbl-works/terraform/pull/322/files#diff-e5a58c2e0c8d4b4f2a5f57fdd0b4d2c084b1b5482f9103cc8c898c9bfd9b7888L2

File: azure/container-app/main.tf [hunk 2]
```diff
@@ -21,9 +21,24 @@ resource "azurerm_role_assignment" "main" {
   principal_id         = data.azurerm_user_assigned_identity.main.principal_id
 }
 
+# https://learn.microsoft.com/en-us/azure/container-apps/vnet-custom?tabs=bash%2Cazure-cli&pivots=azure-portal
+resource "azurerm_container_app_environment" "main" {
+  name                = local.default_name
```
[the remark this answers is not in the document]
Comment (swiknaba) [reply azure/container-app/main.tf]: @samkahchiin ping

File: azure/virtual-network/db-subnet.tf (hunk 1 above)
Comment (swiknaba) [reply azure/virtual-network/db-subnet.tf:22]: @samkahchiin ping

File: azure/container-app/variables.tf [hunk 3]
```diff
@@ -75,7 +75,7 @@ locals {
 variable "revision_mode" {
   type        = string
   description = "In Single mode, a single revision is in operation at any given time. In Multiple mode, more than one revision can be active at a time and can be configured with load distribution via the traffic_weight block in the ingress configuration."
-  default     = "Multiple"
+  default     = "Single"
```
Comment (swiknaba) [inline azure/container-app/variables.tf:78]: don't we need "Multiple" for zero-downtime deploys?

Comment (samkahchiin) [reply azure/container-app/variables.tf:78]:
```
In single revision mode, Container Apps ensures your app doesn't experience downtime when creating a new revision. The existing active revision isn't deactivated until the new revision is ready. 
```
https://learn.microsoft.com/en-us/azure/container-apps/revisions

Comment (swiknaba) [reply azure/container-app/variables.tf:78]: awesome! Can we add to the description, that single mode is enough for zero-downtime deployments?

**Review by swiknaba — APPROVED** [review]

**Pull request merged**